Bugsnag/KSCrash/Source/KSCrash/Recording/Tools/BSG_KSString.c

```diff
@@ -132,7 +132,7 @@ bool bsg_ksstring_extractHexValue(const char *string, size_t stringLength,
                 accum += nybble;
             }
             *result = accum;
-            return true;
+            return true; //!OCLINT
```

Comment:
Is this a rule about early returns or something? Is that something that should be generally adopted?

Reply:
Actually the code itself is kind of ugly; I don't think it's flagged just because of the early return. It could probably be written in a much clearer form that also wouldn't trigger a lint warning (in this case it's probably the return at the end of a for loop block).

Reply:
The message from oclint in this case was "avoid branching statement as last in loop"

Reply:
Yeah, that's definitely from my weird-ass for loop: https://github.com/bugsnag/bugsnag-cocoa/blob/7295d66f49217de2ff639f013d984a0736a0da9b/Bugsnag/KSCrash/Source/KSCrash/Recording/Tools/BSG_KSString.c#L117

It's basically using breaks and continues and then does a return if it reaches the end of the block. It could definitely be written in a clearer, more standard way.

Reply:
This function has now been refactored